### PR TITLE
Only show assignee learners on Exam Report Page

### LINKED ISSUE #4165: Exam reports will always show progress for entire class/every group, regardless of recipient assignment
### Observed behavior
An exam assigned to 1 group, will show reports for every group
An exam assigned to 2+ groups, will show reports for every group

### Expected behavior
An exam assigned to 1 group will only show reports for that one grouop
An exam assigned to 2+ groups will only show reports designated to those number of groups
An exam assigned to "entire class" will show reports for the entire class

Proper behavior shown in invision prototype: https://invis.io/MKNL93TAZ2E#/276734704_Exam_Progress_One_Group

### User-facing consequences
- Does not follow specced design
- Shows extraneous data and information that doesn't pertain to the recipients
- Harder to find and sort information for that particular group(s)
- Misleading. The report is showing data for groups/learners that I have no intention to assign the exam to

### Steps to reproduce
- Create a fresh exam
- Change recipients from 'Entire class' to a specific number of groups (1 group or 2, doesn't matter)
- Check the exam reports 
- When "View by groups" is unchecked, see that progress for every learner shows
- When "View by groups" is checked, see that progress for every group shows

### Context
 * Kolibri version: 10.1
 * Operating system: MacOS
 * Browser: Chrome

Screenshots:
screen shot 2018-08-17 at 10 21 14 am

Exam is designated for 3 groups

screen shot 2018-08-17 at 10 21 27 am

Exam actually shows every group

### PR DESCRIPTION
**Pull request by jonboiser**
### Summary

1. In Exam report page, filter for students who are currently assigned the exam
1. Add gherkin stories for Exam Report page
1. Refactor ExamREportPage component

**Example list of learners**
screen shot 2018-09-27 at 10 49 06 am

**Example list of groups**
screen shot 2018-09-27 at 10 49 12 am

**Exam with no assignees**
screen shot 2018-09-27 at 10 48 46 am

**Exam with some assignees**
screen shot 2018-09-27 at 10 49 37 am

### Reviewer guidance

1. Test with exams that are assigned to whole class
1. Test with exams that are assigned to groups (but total assignees = 0)
1. Test with exams that are assigned to groups (but groups != whole class)

### References

Fixes #4165 

----

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

**Review by khangmach — APPROVED** [review]
LGTM

File: kolibri/plugins/coach/assets/src/modules/examReport/index.js [hunk 1]
```diff
@@ -14,6 +15,23 @@ function defaultState() {
 export default {
   namespaced: true,
   state: defaultState(),
+  getters: {
+    learnerIsExamAssignee(state) {
+      const { assignments } = state.exam;
+      return function isAssigned(learner) {
+        // If assigned to whole class, then true
+        if (assignments.length === 1 && assignments[0].collection_kind === 'classroom') {
```
Comment (christianmemije) [inline kolibri/plugins/coach/assets/src/modules/examReport/index.js]: Reference the constant

File: kolibri/plugins/coach/assets/src/views/exams/ExamReportPage/index.vue [hunk 2]
```diff
@@ -137,46 +121,44 @@
...
-      examKind() {
-        return ContentNodeKinds.EXAM;
-      },
       actionOptions() {
         return [
-          { label: this.$tr('previewExam') },
-          { label: this.$tr('editDetails') },
-          { label: this.$tr('copyExamOptionLabel') },
-          { label: this.$tr('delete') },
+          { label: this.$tr('previewExam'), value: ExamModals.PREVIEW_EXAM },
```
Comment (christianmemije) [inline kolibri/plugins/coach/assets/src/views/exams/ExamReportPage/index.vue:158]: Clever

**Review by christianmemije — APPROVED** [review]
10/10 Looks good

**Pull request merged**